### PR TITLE
Feature 17 droop

### PR DESCRIPTION
**Pull request by Webbah**
Extended safeopt examples:
- Single voltage forming inverter, controller layout using safeopt
- Voltage forming and current sourcing inverter: droop controller layout using safeopt

Corrected xlabel scaling in env render

Bug fix in safeopt agent in case of episode abortion 

Added loadstep using partial

### CODE REVIEW

File: examples/single_inverter_voltage_current_control_safe_opt.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+            self._idx = nested_map(
+                lambda n: obs.index(n),
+                [[f'lc1.inductor{k}.i' for k in '123'], 'master.phase', [f'master.SPI{k}' for k in 'dq0'],
+                 [f'lc1.capacitor{k}.v' for k in '123'], [f'master.SPV{k}' for k in 'dq0']])
+
+    def rew_fun(self, cols: List[str], data: np.ndarray) -> float:
+        """
+        Defines the reward function for the environment. Uses the observations and setpoints to evaluate the quality of
+        the used parameters.
+        Takes current and voltage measurement and setpoints to calculate the mean-root-error control error and uses a
```
Comment (wallscheid) [inline examples/single_inverter_voltage_current_control_safe_opt.py:55]: 2xerror

File: examples/single_inverter_voltage_current_control_safe_opt.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    # Define the droop parameters for the inverter of the active power Watt/Hz (DroopGain), delta_t (0.005) used for the
+    # filter and the nominal frequency
+    # Droop controller used to calculate the virtual frequency drop due to load changes
+    droop_param = DroopParams(DroopGain, 0.005, nomFreq)
+
+    # Define the Q-droop parameters for the inverter of the reactive power VAR/Volt, delta_t (0.002) used for the
+    # filter and the nominal voltage
+    qdroop_param = DroopParams(QDroopGain, 0.002, nomVoltPeak)
+
+    # Define a voltage forming inverter using the pi pi and droop parameters from above
```
Comment (wallscheid) [inline examples/single_inverter_voltage_current_control_safe_opt.py:135]: ....PIPI control and drop parameters....

File: examples/single_inverter_voltage_current_control_safe_opt.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+    #####################################
+    # Execution of the experiment
+    # Using a runner to execute 'num_episodes' different episodes (i.e. SafeOpt iterations)
+    runner = Runner(agent, env)
+
+    runner.run(num_episodes, visualise=True)
+
+    #####################################
+    # Performance results and Parameters as well as plots are stored in folder pipi_signleInv
```
Comment (wallscheid) [inline examples/single_inverter_voltage_current_control_safe_opt.py:225]: parameters

File: examples/single_inverter_voltage_current_control_safe_opt.py (hunk 1 above)
Comment (wallscheid) [inline examples/single_inverter_voltage_current_control_safe_opt.py:55]: 2xerror

File: examples/two_inverter_droop_safe_opt.py [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+            self._idx = nested_map(
+                lambda n: obs.index(n),
+                [[f'slave.freq'],
+                 [f'master.CVV{s}' for s in 'dq0']])
+
+    def rew_fun(self, cols: List[str], data: np.ndarray) -> float:
+        """
+        Defines the reward function for the environment. Uses the observations and setpoints to evaluate the quality of
+        the used parameters.
+        Takes current measurement and setpoints so calculate the mean-root-error control error
```
Comment (wallscheid) [inline examples/two_inverter_droop_safe_opt.py:66]: 2xerror

File: examples/single_inverter_voltage_current_control_safe_opt.py (hunk 2 above)
Comment (Webbah) [reply examples/single_inverter_voltage_current_control_safe_opt.py:135]: done

**Pull request merged**